In [77]:
import pandas as pd
from sklearn.model_selection import train_test_split
from sklearn.naive_bayes import GaussianNB
from sklearn.metrics import classification_report

# Load the glass dataset
glass_df = pd.read_csv("glass.csv")

# Split the data into training and testing sets
X = glass_df.drop("Type", axis=1)
y = glass_df["Type"]
X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.25)

# Train the Gaussian Naive Bayes model
gnb_model = GaussianNB()
gnb_model.fit(X_train, y_train)

# Predict on the test set
y_pred = gnb_model.predict(X_test)

# Evaluate the model
gnb_svc = round(gnb_model.score(X_train, y_train) * 100, 2)

print("Accuracy:", gnb_svc)
print("Classification Report:\n", classification_report(y_test, y_pred , zero_division=1))


Accuracy: 60.62
Classification Report:
               precision    recall  f1-score   support

           1       0.39      0.87      0.54        15
           2       0.80      0.18      0.30        22
           3       0.40      0.29      0.33         7
           5       1.00      0.67      0.80         3
           6       0.75      0.75      0.75         4
           7       0.60      1.00      0.75         3

    accuracy                           0.50        54
   macro avg       0.66      0.63      0.58        54
weighted avg       0.63      0.50      0.46        54



Here imported pandas to load DataFrame to glass_df variable, then splits the dataframe to train and test data sets with test size of 0.25.

The Gaussian Naive Bayes model is trained on the training data using the fit method. After training, the model is used to make predictions on the test set using the predict method.

Finally, the accuracy of the model is evaluated using the score method and a classification report is generated using the classification_report method. The accuracy score and the classification report give an insight into the performance of the model on the test set.

In [84]:
import pandas as pd
from sklearn.model_selection import train_test_split
from sklearn.svm import LinearSVC
from sklearn.metrics import classification_report

# Load the glass dataset
glass_df = pd.read_csv("glass.csv")

# Split the data into training and testing sets
X = glass_df.drop("Type", axis=1)
y = glass_df["Type"]
X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.25)

# Train the linear SVM model
svm_model = LinearSVC()
svm_model.fit(X_train, y_train)

# Predict on the test set
y_pred = svm_model.predict(X_test)

# Evaluate the model
svm_accuracy = round(svm_model.score(X_train, y_train) * 100, 2)
print("Accuracy:", svm_accuracy)

print("Classification Report:\n", classification_report(y_test, y_pred , zero_division=1))


Accuracy: 37.5
Classification Report:
               precision    recall  f1-score   support

           1       1.00      0.00      0.00        19
           2       0.43      0.91      0.58        23
           3       0.00      0.00      0.00         3
           5       1.00      0.00      0.00         1
           6       0.00      0.00      0.00         1
           7       1.00      0.14      0.25         7

    accuracy                           0.41        54
   macro avg       0.57      0.18      0.14        54
weighted avg       0.68      0.41      0.28        54



C:\Users\bhuva\anaconda3\lib\site-packages\sklearn\svm\_base.py:1206: ConvergenceWarning: Liblinear failed to converge, increase the number of iterations.
  warnings.warn(


Here imported pandas to load DataFrame to glass_df variable, then splits the dataframe to train and test data sets with test size of 0.25.

The SVC model is trained on the training data using the fit method. After training, the model is used to make predictions on the test set using the predict method.

Finally, the accuracy of the model is evaluated using the score method and a classification report is generated using the classification_report method. The accuracy score and the classification report give an insight into the performance of the model on the test set.

## Which algorithm you got better accuracy? Can you justify why?

From the results, it looks like the Gaussian Naive Bayes algorithm has a higher accuracy compared to the linear SVM  Gaussian Naive Bayes algorithm. In this case, the SVM model has an accuracy of 37.5% while the Gaussian Naive Bayes algorithm has an accuracy of 60.62%.
